client/layout/index.jsx

```diff
@@ -151,10 +153,13 @@ class Layout extends Component {
 				<AsyncLoad require="layout/guided-tours" placeholder={ null } />
 				{ ! isE2ETest() && <AsyncLoad require="layout/nps-survey-notice" placeholder={ null } /> }
 				{ config.isEnabled( 'keyboard-shortcuts' ) ? <KeyboardShortcutsMenu /> : null }
-				<MasterbarLoggedIn
-					section={ this.props.sectionGroup }
-					isCheckout={ this.props.sectionName === 'checkout' }
-				/>
+				{ 'jetpack-cloud' === this.props.application && <JetpackMasterbar /> }
```

Comment:
We should try to reuse the same masterbar component if possible, and have its behaviour altered by feature flags, e.g. whether to include the Reader. That would allow this file to remain unmodified.

Reply:
Please consider that the current design direction is quite different, so it might make sense, at least for the MVP, to have a very basic condition at the top and styling a simpler bar. We need to have a discussion here together with design.

Reply:
I think keeping these separate would be a better approach for reasons @folletto says. Would be good to find solid common boundaries at the component level but re-using the whole thing across two quite different applications is going to get messy in code already to begin with (lots of flags altering visuals/mechanics) and eventually block or slow down any significant refactoring at Calypso side. We have plans at .com to do visual/mechanics refactoring soonish and I'd prefer it to be not any harder project than it already is.